openhcl/underhill_core/src/dispatch/mod.rs

```diff
@@ -660,6 +649,12 @@ impl LoadedVm {
             .map(vmcore::save_restore::SaveRestore::save)
             .transpose()
             .context("shared_vis_pool save failed")?;
+        let private_pool = self
```

Comment:
So the difference is that we're saving it unconditionally as long as it exist.

Reply:
right, which i think we need to do. We still need to fix the drivers to free their allocations before saving this though, if they use this pool.